### PR TITLE
Calculator-Store user's previous marks

### PR DESCRIPTION
**Pull request by vgoyal1996**
Added functionality to CalculatorActivity.java to store and retrieve user's previous marks. Marks are stored in the form of a text file in phone's memory. Marks are stored in the file when the user hits the "Calculate" button. Option for retrieving previous marks is provided in the overflow menu. If marks for the chosen semester and branch are available, they are retrieved otherwise, a "marks not available" message is given. Sorry for the misleading branch name :)

### CODE REVIEW

**Comment by Swati4star** [comment]
Hi @vipulabhay1996 Can you please work on why the checks are failing?

Also a suggestion, improve the future commit messages, and mention what you are actually changing in that commit.

**Comment by vgoyal1996** [comment]
@Swati4star Thank you for the suggestion. Will keep in mind.
Can you please clarify what checks are you referring to?

**Comment by Swati4star** [comment]
Travis Check. You must be able to see the message 'All checks have failed'
Check that.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -154,6 +154,8 @@
     <string name="st11" />
     <string name="st12" />
     <string name="percentage" />
+    <string name="action_show_previous_marks">show previous marks</string>
```
Comment (Swati4star) [inline app/src/main/res/values/strings.xml]: change it to 'Show previous marks'

S capital

File: app/src/main/java/nsit/app/com/nsitapp/CalculatorActivity.java [hunk 2]
```diff
@@ -14,14 +18,27 @@
...
+import static nsit.app.com.nsitapp.R.id.per;
+
 public class CalculatorActivity extends Fragment {
 
     private Spinner s1,s2;
-    private TextView tc10, tc11, tc12, tc9, tc4, tc5;
+    private TextView tc10, tc11, tc12, tc9, tc4, tc5, tc1,tc2,tc3,tc8,tc7,tc6,rs;
+    private double n1=-1.0, n2=-1.0, n3=-1.0, n4=-1.0, n5=-1.0, n6=-1.0, n7=-1.0, n8=-1.0, n9=-1.0, n10=-1.0, n11=-1.0, n12=-1.0;
+    private EditText t1,t2,t3,t4,t5,t6,t7,t8,t9,t10,t11,t12;
+
```
Comment (Swati4star) [inline app/src/main/java/nsit/app/com/nsitapp/CalculatorActivity.java]: Change the name of all the variables to something appropriate. Something these variables signify.

**Comment by vgoyal1996** [comment]
@Swati4star changed the variable names.
please check.

**Comment by Swati4star** [comment]
Don't close and reopen the PR again and again.

I have added one more change.

File: app/src/main/java/nsit/app/com/nsitapp/CalculatorActivity.java [hunk 3]
```diff
@@ -42,263 +72,276 @@ public void onClick(View v) {
...
 
-                    } else if (s2.getSelectedItem().toString().equals("Sem 5")) {
+                    } else if (semSpinner.getSelectedItem().toString().equals("Sem 5")) {
 
-                        tc10.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc9.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc12.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                    } else if (s2.getSelectedItem().toString().equals("Sem 6")) {
+                        PR5textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                        PR4textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
```
Comment (Swati4star) [inline app/src/main/java/nsit/app/com/nsitapp/CalculatorActivity.java]: This line `textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));` is being repeated a lot. Why don't you create a function that do this.

A function that takes 2 arguments : textview and color, and performs the task.

**Comment by Swati4star** [comment]
Also, a suggestion, use [Butterknife](https://github.com/JakeWharton/butterknife) in the project. We would eventually be doing this later.

**Comment by vgoyal1996** [comment]
@Swati4star please review the changes.

**Comment by Swati4star** [comment]
:+1: Great @vipulabhay1996 I will once test the app by running the code. Changes seems fine.

**Comment by vgoyal1996** [comment]
@Swati4star any updates?

**Comment by Swati4star** [comment]
merged :+1: Thanks!

**Pull request closed without merging**